### PR TITLE
docs: Explain Python usage

### LINKED ISSUE #1502: Improve developer documentation examples
### User Story

So that easily write some code to use the geostore, as a python developer, I want some examples on how to use the Geostore.

#### Acceptance Criteria

- [ ] All example code and commands in the Geostore USAGE.md also have a python example

#### Tasks

- [ ] ...
- [ ] ...

#### Definition of Ready

- [ ] This story is **ready** to work on
  - [ ] Independent (story is independent of all other tasks)
  - [ ] Negotiable (team can decide how to design and implement)
  - [ ] Valuable (from a user perspective)
  - [ ] Estimate value applied (agreed by team)
  - [ ] Small (so as to fit within an iteration)
  - [ ] Testable (in principle, even if there isn't a test for it yet)
  - [ ] Environments are ready to meet definition of done
  - [ ] Resources required to implement will be ready
  - [ ] Everyone understands and agrees with the tasks to complete the story
  - [ ] Release value (e.g. Iteration 3) applied
  - [ ] Sprint value (e.g. Aug 1 - Aug 15) applied

#### Definition of Done

- [ ] This story is **done**:
  - [ ] Acceptance criteria completed
  - [ ] Automated tests are passing
  - [ ] Code is peer reviewed and pushed to master
  - [ ] Deployed successfully to test environment
  - [ ] Checked against [CODING guidelines](https://github.com/linz/geostore/blob/master/CODING.md)
  - [ ] Relevant new tasks are added to backlog and communicated to the team
  - [ ] Important decisions recorded in the issue ticket
  - [ ] Readme/Changelog/Diagrams are updated
  - [ ] Product Owner has approved acceptance criteria as complete
  - [ ] Meets non-functional requirements:
    - [ ] Scalability (data): Can scale to 300TB of data and 100,000,000 files and ability to
          increase 10% every year
    - [ ] Scability (users): Can scale to 100 concurrent users
    - [ ] Cost: Data can be stored at < 0.5 NZD per GB per year
    - [ ] Performance: A large dataset (500 GB and 50,000 files - e.g. Akl aerial imagery) can be
          validated, imported and stored within 24 hours
    - [ ] Accessibility: Can be used from LINZ networks and the public internet
    - [ ] Availability: System available 24 hours a day and 7 days a week, this does not include
          maintenance windows < 4 hours and does not include operational support
    - [ ] Recoverability: RPO of fully imported datasets < 4 hours, RTO of a single 3 TB dataset <
          12 hours

### PR DESCRIPTION
**Pull request by l0b0**
## Issues

Closes #1502.

## Reference

[Code review checklist](CODING.md#Code-review-checklist)

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -150,7 +150,9 @@ Example of assuming the Geostore roles from a LINZ AWS account:
 
 You can communicate with a Geostore instance using either the low-level application programming
 interface (API) or the high-level command-line interface (CLI). The following documentation explains
-how to work with both.
+how to work with both. For an example of working with Python, see the
+`should_successfully_run_dataset_version_creation_process_with_multiple_assets` function in
```
Comment (billgeo) [inline USAGE.md:154]: This is quite a big function which looks like most of it is creating dummy files. Is there a way to publish a URL to just the relevant bits showing how to call the lambda functions with python? Or a GIST or something?

**Review by paulfouquet — APPROVED** [review]

File: USAGE.md (hunk 1 above)
Comment (paulfouquet) [reply USAGE.md:154]: I agree for the visibility but the good thing about pointing to this unit test is that if the function call change, the test will have to be updated. Might be more difficult to notice if the example is somewhere else?

Comment (l0b0) [reply USAGE.md:154]: You're right that it's big, but it's not creating "dummy" files. The files it creates are the dataset. So the test basically shows the whole life cycle from creating a dataset in a staging bucket, calling the Geostore, verifying that the Geostore imported everything correctly, and then tearing down the contents of the staging bucket. Any separate code, as Paul points out, will eventually get out of date.

Comment (billgeo) [reply USAGE.md:154]: Okidoke. If you can find what you need in there @paulfouquet . All good. 🚀

**Review by billgeo — APPROVED** [review]

**Pull request merged**